src/stream.rs

```diff
@@ -11,7 +11,7 @@ use openssl::x509::X509_FILETYPE_PEM;
 
 /// Encapsulates the functionality for how to connect to the server.
 #[derive(Clone)]
-pub enum StreamConnector {
+pub enum ConnectMethod {
```

Comment:
Because `ClientOptions` was getting renamed to `Connector`, I felt that this type required a change as well in order to be less confusing. I'm not completely convinced that `Connector` and `ConnectMethod` are the best names we can come up with though, so suggestions are welcome.

Additional changes beyond the renaming are from rustfmt.